src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java

```diff
@@ -174,28 +164,101 @@ public byte[] mask(byte[] input) {
     }
 
     /**
-     * Checks if the current byte changes the current json path. If so, updates current json path.
+     * Masks the array element in place.
+     *
+     * @param maskingState the masking state with the index on the element prior to the element start index
+     */
+    private void maskArrayElementInPlace(MaskingState maskingState, KeyMaskingConfig keyMaskingConfig) {
+        maskingState.incrementCurrentIndex();
+        skipWhitespaceCharacters(maskingState);
+        boolean maskable = maskValue(maskingState, keyMaskingConfig);
+        if (!maskable) {
+            // non-maskable values
+            skipAllValues(maskingState);
+        }
```

Comment:
Is this only applied for `nulls` and when number / boolean masking is disabled? If I understand correctly for objects and arrays we always skip values inside `maskArrayValueInPlace` and `maskObjectValueInPlace`. I'd rather move this logic inside `mask*ValueInPlace`, otherwise it's slightly confusing

Reply:
`maskValue` would return false only for null values. I created this method because sonarqube failed on code duplication as I used the same sequence of "if-else" statements to check the value type as in `maskObjectValueInPlace` method. The methods that use `maskValue` handle non-maskable values differently, so I cannot move the handling to `maskValue` method.

Reply:
Wouldn't it make sense to move `skipAllValues` inside `maskValue` then? I've tried [this](https://github.com/Breus/json-masker/files/14421714/maskValue_refactor.patch) and all tests seem to pass.

Reply:
If the tests pass then let's contain the skipping logic inside `maskValue`.

Reply:
Done